### PR TITLE
Force retraining of outdated models

### PR DESCRIPTION
**Pull request by pheel**
Since [this](https://github.com/RasaHQ/rasa/pull/6276), Rasa is able to boot when detected model is outdated. Since [this](https://github.com/RasaHQ/rasa/blob/cc8a2647604cfc96138813c3081d5268af622c10/rasa/train.py#L263), Core-only trainings are able to fall back on using an interpreter from the zipped model. If the zipped NLU model is outdated, Rasa will not train.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @b-quachtran will take a look at it as soon as possible ✨

**Comment by pheel** [comment]
@alwx 🙏

**Review by alwx — APPROVED** [review]
Thank you for your contribution and sorry that it took me so much time to get it reviewed.

The code looks good. I was thinking that it might make sense to add a test for outdated models and checking if the model is actually detected as the outdated (it'd be easier to just move `model_outdated` to a separate function and add a small test for this function)

However, it's a relatively small thing, so I am going to approve this PR anyway.

**Comment by pheel** [comment]
Pinging @alwx in case...

File: tests/test_model.py [hunk 1]
```diff
@@ -128,8 +129,8 @@ def _fingerprint(
     nlg: Optional[Any] = None,
     stories: Optional[Any] = None,
     nlu: Optional[Any] = None,
-    rasa_version: Text = "1.0",
-) -> Fingerprint:
+    rasa_version: Text = MINIMUM_COMPATIBLE_VERSION,
+):
```
Comment (wochinge) [inline tests/test_model.py:133]: can you please re-add the removed return type?

**Review by wochinge — COMMENTED** [review]
How did you train the second model? Via `rasa train core`? And you could please add a test to verify the bug was fixed (that also helps me to understand the bug better).

Thanks for taking the initiative for the fix!

**Comment by wochinge** [comment]
@pheel Please feel free to re-open if you have time to continue

**Pull request closed without merging**